### ISSUE TITLE
Implement `VK_EXT_host_image_copy`

### ISSUE THREAD

**Issue by DUOLabs333** [KhronosGroup/MoltenVK#2113]
It's not a complete blocker, but it would make copying images to memory needed (instead of having to go through the whole VkBuffer setup).

**Comment by spnda** [comment]
This should definitely be doable without too much work as Metal's texture object provides effectively the same exact API for working with textures.

The only difference which can easily by solved with a `MTLBlitCommandEncoder` that we submit directly is the `vkCopyImageToImageEXT` function. Metal only allows copy from one texture to another through command buffers. I'm not exactly sure about the sync implications of using a command encoder in that function, but I think it's fine since the API user is required to sync the image accesses themselves.

As `vkCopyMemoryToImageEXT` and `vkCopyImageToMemoryEXT` both allow copying to and from DEVICE_LOCAL only memory, there are some changes needed. This is fine for Apple Silicon GPUs, as we can just force StorageModeShared on those GPUs, as there will be no disadvantage of doing so. For dedicated GPUs without shared memory, this could get a bit trickier. Perhaps the extension is only something we can support on Apple Silicon GPUs.

I'm on implementing this now.

**Comment by billhollings** [comment]
> Perhaps there's some way we can somehow force `StorageModeShared` for resources with `VK_FORMAT_FEATURE_2_HOST_IMAGE_TRANSFER_BIT_EXT`

Yes. I'd say if `VK_FORMAT_FEATURE_2_HOST_IMAGE_TRANSFER_BIT_EXT` is set when the image is created, we should force either Shared or Managed memory.

For the actual copying, consider using `MVKImagePlane::getMTLTextureContent()` and `MVKImagePlane::updateMTLTextureContent`. And I expect arranging them back-to-back should work for `vkCopyImageToImageEXT()` too, without the need for a command-buffer-based BLIT command.
